cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,189 @@
+    select_volume.select_by_value(v_option[1])
+    pytest_selenium.click(submit_changes)
+    assert_message_contain('Detaching Cloud Volume')
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    info_el = InfoBlock.element('Relationships', 'Cloud Volumes')
+    pytest_selenium.click(info_el)
+    res = list_tbl.find_cell('Name', v_option[0])
+    assert not res, 'Volume does not disappear from instance relationships'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_attach_volume_from_volume_page(provider):
+    """Attaches pre-created volume to an instance from Volume page"""
+    vms = filter(lambda vm: vm.power_state == 'ACTIVE', provider.mgmt.all_vms())
+    inst_name = choice(vms).name
+    navigate_to(Volume, 'All')
+    params = {'Status': 'available',
+              'Cloud Provider': provider.get_yaml_data()['name']}
+    list_tbl.click_row_by_cells(params, 'Name')
+    vname = get_volume_name()
+    toolbar.select('Configuration', 'Attach this Cloud Volume to an Instance')
+    assert len(select_vm.all_options) > 0
+
+    select_vm.select_by_value(select_vm.get_value_by_text(inst_name))
+    pytest_selenium.send_keys(device_input, '/dev/vdb')
+    pytest_selenium.click(attach_volume)
+    assert_message_contain('Attaching Cloud Volume')
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    navigate_to(Volume, 'All')
+    params = {'Name': vname, 'Status': 'in-use'}
+    res = list_tbl.find_rows_by_cells(params)
+    assert res, 'Volume does not marked as in-use'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_detach_volume_from_volume_page(provider):
+    """Detaches volume from instance from Volume page"""
+    navigate_to(Volume, 'All')
+    params = {'Status': 'in-use',
+              'Cloud Provider': provider.get_yaml_data()['name']}
+    list_tbl.click_row_by_cells(params, 'Name')
+    vname = get_volume_name()
+    toolbar.select('Configuration', 'Detach this Cloud Volume from an Instance')
+    assert len(select_vm.all_options) > 0
+    select_vm.select_by_index(1)
+    pytest_selenium.click(detach_volume)
+    assert_message_contain('Detaching Cloud Volume')
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    navigate_to(Volume, 'All')
+    params = {'Name': vname, 'Status': 'available'}
+    res = list_tbl.find_rows_by_cells(params)
+    assert res, 'Volume does not marked as available'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_delete_volume(provider):
+    """Deletes volume"""
+    navigate_to(Volume, 'All')
+    params = {'Status': 'available',
+              'Cloud Provider': provider.get_yaml_data()['name']}
+    list_tbl.click_row_by_cells(params, 'Name')
+    vname = get_volume_name()
+    toolbar.select('Configuration', 'Delete this Cloud Volume',
+                   invokes_alert=True)
+    pytest_selenium.handle_alert(cancel=False)
+    assert_message_contain('Delete initiated for 1 Cloud Volume.')
```

Comment:
We usually import the flash module and not the particular functions. `flash.assert_message_contain` is more obvious than just `assert_message_contain`.